proposals/lip-0017.md

```diff
@@ -89,105 +93,133 @@ There are, however, still some limitations that disallow arbitrarily large numbe
+           validOptionalSig += 1
+       else if signatures[numMandatoryKeys + k].byteLength != 0
+           return false
+
+   if numMandatorKeys + validOptionalSig == numberOfSignatures
+       return true
+   else
+       return false
+```
 
-    The `signatures` property of the current protocol which holds an array of signatures is replaced by this one.
+Therefore, for transactions outgoing from multisignature accounts, `signatures` must be of length `length(mandatoryKeys) + length(optionalKeys)`.
 
-Moreover, the `signature` property is ignored in the transaction verification.
 
 ### Transaction Broadcasting
 
-When a node receives a transaction, it first has to verify the transaction before broadcasting. This includes the steps 1 and 2 from the [section above](#transaction-creation-and-verification) for transactions from multisignature accounts. In particular, a node is not allowed to broadcast a transaction that does not have the correct number of signatures (step 2).
+When a node receives a transaction, it first has to verify the transaction before broadcasting. This includes the steps from the [section above](#transaction-creation-and-verification) for transactions from multisignature accounts. In particular, a node is not allowed to broadcast a transaction that does not have the appropriate signatures.
 
 ### Transaction Serialization
 
 #### Asset Property Serialization for Multisignature Registration Transactions
 
-The asset property of a transaction of type 4 has to be serialized as by the function `serialize_asset_for_type_4` in the pseudocode below. This serialization is used to compute the byte array of a transaction for the signature computation, the transactionID computation and to include the transaction into the payload hash of a block.
+The asset property of a Multisignature Registration Transaction has to be serialized as by the function `serialize_asset_for_multisignature_registration` in the pseudocode below. This serialization is used to compute the byte array of a transaction for the signature computation, for the ID computation and to include the transaction into the payload hash of a block.
 
 ```
-serialize_asset_for_type_4(asset):
-    let byteBuffer be an empty byte buffer
-    let lm be the length of the array asset.mandatoryKeys
-    append lm as an 8-bit unsigned integer to byteBuffer
-    for each key in asset.mandatoryKeys:
-        append the big endian encoding of key to byteBuffer
-    let lo be the length of the array asset.optionalKeys
-    append lo as an 8-bit unsigned integer to byteBuffer
-    for each key in asset.optionalKeys:
-        append the big endian encoding of key to byteBuffer
-    append asset.numberOfSignatures as an 8-bit unsigned integer to byteBuffer
-    return byteBuffer
+serialize_asset_for_multisignature_registration(asset):
+   let byteBuffer be an empty byte buffer
+   let lm be the length of the array asset.mandatoryKeys
```

Comment:
as we're using some variables in this pseudo code with full words I would do the same for `lm` and `lo` that probably mean `lengthMandatory` and `lengthOptional`

Reply:
changed